mkosi

```diff
@@ -2406,11 +2419,18 @@ def setup_package_cache(args: CommandLineArguments) -> Optional[tempfile.Tempora
     return d
 
 class ListAction(argparse.Action):
+    def __init__(self, *args, choices=None, **kwargs):
+        self.list_choices = choices
+        super().__init__(*args, **kwargs)
     def __call__(self, parser, namespace, values, option_string=None):
         l = getattr(namespace, self.dest)
         if l is None:
             l = []
-        l.extend(values.split(self.delimiter))
+        new = values.split(self.delimiter)
+        for x in new:
+            if self.list_choices is not None and x not in self.list_choices:
+                raise ValueError('Unknown value {!r}'.format(x))
+            l.append(x)
```

Comment:
not that it matters, but it would be sexier to continue to use l.extend() if lis_choices is None... I mean, Python is not the fastest language in the world and needless loops sound needless to me ;-)

Reply:
Nah, it's in option parsing code, and a few ten or hundreds of microseconds don't matter here. This runs just once.